### PR TITLE
Generate a self signed certificate when flight-www is installed

### PR DESCRIPTION
**Pull request by WilliamMcCumstie**
`HTTPs` is enabled implicitly when `flight-www` by generating a self signed certificate. The `domain` defaults to `hostname --fqdn`.

The `flight www cert-gen` command can be used to convert it into a `lets-encrypt` certificate.

### CODE REVIEW

File: builders/flight-www/package-scripts/flight-www/postinst [hunk 1]
```diff
@@ -49,4 +49,21 @@ sed -i -e '/cron_path/s/daily/weekly/' /opt/flight/etc/share/cert.yaml
 EOF
 fi
 
+# Generate a self-signed certificate implicitly
+if [ ! -f "/opt/flight/etc/www/ssl/key.pem" ]; then
+  domain=$(hostname --fqdn)
```
Comment (benarmston) [inline builders/flight-www/package-scripts/flight-www/postinst:54]: This exactly what I requested on the trello card.  I now realise that I meant `hostname -d`.  Can you change this please?

File: builders/flight-www/package-scripts/flight-www/postinst [hunk 2]
```diff
@@ -49,4 +49,21 @@ sed -i -e '/cron_path/s/daily/weekly/' /opt/flight/etc/share/cert.yaml
...
+  /opt/flight/bin/flight www cert-gen --domain "$domain" --cert-type self-signed >/dev/null 2>&1
+  /opt/flight/bin/flight www enable-https >/dev/null 2>&1
```
Comment (benarmston) [inline builders/flight-www/package-scripts/flight-www/postinst:56]: There isn't any error handling here.  If either of these fail, the message below will still be displayed and mislead the user into believing that HTTPS has been enabled.  We need to only show the message below if HTTPS actually has been enabled.

**Review by benarmston — CHANGES_REQUESTED** [review]
I've pushed a commit changing the message slightly.  I've also added a couple of comments for you to look at.

File: builders/flight-www/package-scripts/flight-www/postinst (hunk 2 above)
Comment (WilliamMcCumstie) [reply builders/flight-www/package-scripts/flight-www/postinst]: https://github.com/openflighthpc/openflight-omnibus-builder/pull/125/commits/409f1d01193212c9f4f930960194bf54c35bc73e

File: builders/flight-www/package-scripts/flight-www/postinst (hunk 1 above)
Comment (WilliamMcCumstie) [reply builders/flight-www/package-scripts/flight-www/postinst]: https://github.com/openflighthpc/openflight-omnibus-builder/pull/125/commits/409f1d01193212c9f4f930960194bf54c35bc73e

**Comment by benarmston** [comment]
@WilliamMcCumstie it looks like you force pushed away the change to the error message I'd made.  Can I get them back please?

**Comment by WilliamMcCumstie** [comment]
Sorry wasn't quite finished, added the change back

**Comment by benarmston** [comment]
I've adjusted the copy again.  It now looks good for `yum install flight-www`.  It didn't quite fit with the other copy when `yum install flight-web-suite`.  I've updated the web suite copy accordingly.

I'm going to put this on hold until web suite 2021.4 is released and then I'll rebase on that branch and double check the copy with `yum install flight-web-suite`.

I may also decide that `flight www cert-gen` output should include the domain that the cert was generated for.

**Comment by benarmston** [comment]
I've rebased on latest `master`.  Not sure if the new version number is correct.  I'll reconsider that later.

**Comment by benarmston** [comment]
@WilliamMcCumstie FYI I'm going to merge this to a new branch `user-web-suites-2021.5`.  Having all of the PRs related to the next user/web suite release on a single branch may make my release process a bit easier.

**Comment by WilliamMcCumstie** [comment]
:+1:

**Pull request merged**